### ISSUE TITLE
NoMethodError `errors` for <GenericWork> when adding work to two collections of same single membership type (nurax-pg)

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#5594]
### Descriptive summary

Adding a work to two collections of the same single membership type raises `NoMethodError - undefined method 'errors' for <GenericWork>`

### Expected behavior

When attempting to add a work to a second collection of the same single membership type, after clicking Save, should stay in the editor and see flash message explaining why this is not allowed.

### Actual behavior

When attempting to add a work to a second collection of the same single membership type, after clicking Save, it attempts to stay in the editor but raises an exception...

<img width="838" alt="image" src="https://user-images.githubusercontent.com/6855473/161634415-dfeb34ea-492b-4c8c-88c4-055bdbfc30b7.png">

### Steps to reproduce the behavior

On nurax-pg...

Prerequisite:
1. Create a collection type that is single membership
2. Create two collections of that type

To Reproduce:
1. Dashboard -> Works -> Add new work -> select any work type
2. Fill in required metadata
3. Select tab: Relationships
4. In Add to collection selection, type name of first single membership collection and click button: Add
5. In Add to collection selection, type name of second single membership collection and click button: Add
6. Click button: Save changes

See exception shown in Actual behavior.

### Related work

Issue #5582

**Comment by acklose** [comment]
@jlhardes When I reviewed this, I got a different error message.

NameError in Hyrax::GenericWorksController#update
uninitialized constant ActiveModel::DeprecationHandlingMessageArray 

<img width="944" alt="Nurax-PB_SingleMembership_twice_2022-09-15" src="https://user-images.githubusercontent.com/26173627/190511044-f8b71ad7-7895-4d78-bbee-b5f8071613f8.png">

**Comment by gamontoya** [comment]
This appears to be working on Koppie. I'm not sure if seeing two errors reported to the user is expected behavior, however.

![singleMem2](https://user-images.githubusercontent.com/4314914/190527488-38ca4805-087c-4f48-8204-3210e0cb3366.png)

**Comment by blancoj** [comment]
@gamontoya The issue you were seeing on Koppie, I also saw and @cjcolvar helped me resolve it.  It is happening because you need the latest version of rails running on Koppie.  This is how you get the latest rails installed on Koppie:

> docker exec -it koppie-app-1 /bin/sh
> bundle update rails

The double error, is what I also saw.  I think this is expected behavior, but it would be good to get someone else to confirm that.

**Comment by gamontoya** [comment]
@blancoj I'll update and try again on koppie, thank you.

@jlhardes This is a fail on nurax-pg:

![single](https://user-images.githubusercontent.com/4314914/190810906-ee9d074d-91cc-4c38-a6a9-87c28463cae3.png)

**Comment by gamontoya** [comment]
@blancoj I'm seeing the same double error banner even after updating rails on koppie.

@jlhardes This passes on koppie.

![singlemem](https://user-images.githubusercontent.com/4314914/190831567-45d236dc-40f3-4473-a157-43563d0f5277.png)

**Comment by jlhardes** [comment]
Verified that this is still producing the same error message on nurax-pg. It is also still working but still showing 2 flash error messages on koppie. Reopening and moving to In Progress column on project board for follow-up.

**Comment by alishaevn** [comment]
instructions on how to create a "single membership" collection can be found on [hyrax#4011](https://github.com/samvera/hyrax/issues/4011) under the "Steps to reproduce the behavior" heading in the initial comment.

**Comment by alishaevn** [comment]
@dlpierce which error message do we want to remove? 

the error message above the form is created by [#after_update_error](https://github.com/samvera/hyrax/blob/main/app/controllers/concerns/hyrax/works_controller_behavior.rb#L402-L411) and rendered by [_flash_msg.html.erb](https://github.com/samvera/hyrax/blob/main/app/views/_flash_msg.html.erb).

the error message within the form is created by [#full_collections_errors](https://github.com/samvera/hyrax/blob/main/app/helpers/hyrax/work_form_helper.rb#L204-L206) and rendered by [_form.html.erb](https://github.com/samvera/hyrax/blob/main/app/views/hyrax/base/_form.html.erb#L13)

| hyrax | koppie |
| --- | --- |
| ![image](https://github.com/samvera/hyrax/assets/29032869/1f87f403-8b14-4253-af2f-259d96c1c359) | ![image](https://github.com/samvera/hyrax/assets/29032869/c61f51b0-69b6-4e37-a790-a6d2db75ab1a) |

**Comment by alishaevn** [comment]
> I'm inclined to keep the one rendered by _flash_msg.html.erb, but having it be blank is not ideal.

@dlpierce agreed. idk why the messages are different between the two, but would you like me to move forward with removing the error message within the form? while I'm at it, should I also remove the [entire errors block](https://github.com/samvera/hyrax/blob/main/app/views/hyrax/base/_form.html.erb#L7-L16) from the form since the flash msg also handles errors on this page?

**Comment by gamontoya** [comment]
@alishaevn 

This is the behavior I'm seeing on nurax-pg


![pgNurax1](https://github.com/samvera/hyrax/assets/4314914/9c5699e4-07b7-41c4-b2f8-6f93234d1c64)

**Comment by gamontoya** [comment]
@alishaevn And this is the behavior on `Koppie v4.0.0`

![koppie](https://github.com/samvera/hyrax/assets/4314914/fe61c91e-9a7a-483f-ba41-b2cf4b3ffe32)

**Comment by alishaevn** [comment]
@gamontoya can you check nurax again? the commit sha listed next to the hyrax revision in the footer is the one for the pr related to this work. looks like it was updated yesterday, but I'm unsure if it was before or after you tested. I don't have a login to check for myself either.

if it doesn't work for you on nurax, can you try locally?

<details>
<summary>local hyrax</summary>

<img width="1440" alt="image" src="https://github.com/samvera/hyrax/assets/29032869/f00018ab-d128-49ba-948b-cd2f249e5d8f">
</details>

<details>
<summary>nurax commit sha</summary>

<img width="1162" alt="image" src="https://github.com/samvera/hyrax/assets/29032869/5ac16130-74c3-4b0d-b3f3-6843d2e01860">
</details>

**Comment by gamontoya** [comment]
@alishaevn Yes, I'm on it!

**Comment by gamontoya** [comment]
@alishaevn This is the latest version of `https://pg.nurax.samvera.org/`

![pg-nurax](https://github.com/samvera/hyrax/assets/4314914/639a0bdd-4cdd-4fb8-b567-5dd6c4a2ccbf)

**Comment by gamontoya** [comment]
My local environment is upated:
`* main 014fe61dc [origin/main] Merge pull request #6090 from samvera/5575-item-actions-edit`

Same behavior for Koppie (it works), unlike pg-nurax

**Comment by alishaevn** [comment]
@gamontoya did your local version of hyrax work? I don't have a login for the pg nurax site so I can't do any debugging there.

**Comment by alishaevn** [comment]
samvera slack message: https://samvera.slack.com/archives/CB69H0SHM/p1686857592249439

**Comment by alishaevn** [comment]
I created a login for pg nurax and tested this. I got the same problem. from the screenshot above, and my testing, it appears that the works created on pg nurax are considered to be valkyrie works instead of active fedora ones. when I test the hyrax app locally, the works are active fedora backed. I'm unsure why the mixup is happening on nurax.

@dlpierce any ideas what's causing this inconsistency?

``` bash
# error from pg
...
hyrax (014fe61dcb0a) app/controllers/concerns/hyrax/works_controller_behavior.rb:203:in `update_valkyrie_work'
hyrax (014fe61dcb0a) app/controllers/concerns/hyrax/works_controller_behavior.rb:104:in `update' 
...


____________________________

# code the above error is referencing
def update
  case curation_concern
  when ActiveFedora::Base
    # this code gets run locally for hyrax
    actor.update(actor_environment) ? after_update_response : after_update_error(curation_concern.errors)
  else
    # this code gets run in pg for hyrax
    update_valkyrie_work
  end
end
```

**Comment by alishaevn** [comment]
as info, I created two collections as "single membership" on https://dev.nurax.samvera.org and it didn't error at all. I'm not an admin, so I can't see how the collection type is truly configured. I was just going by the name. feels like it is not configured correctly though.

**Comment by gamontoya** [comment]
@alishaevn Hi again!

Here's the results from 4 different environments.

**dev-nurax (works)**
-    QA Testing for Hyrax
-    Hyrax Version: 4.0.0
-    Hyrax Branch: refs/heads/main
-   Hyrax Revision: `8aa6379a690945148dbb44e518a0037d370a35e1`

![dev-nurax](https://github.com/samvera/hyrax/assets/4314914/5c151f95-5794-4831-94eb-f33ace5bbe4e) 

**pg-nurax (fails)**
- QA Testing for Hyrax
- Hyrax Version: 4.0.0
- Hyrax Branch: refs/heads/main
- Hyrax Revision: `014fe61dcb0a92f83bbe0b1b2c0828147ec4b4f8`

![pg-nurax](https://github.com/samvera/hyrax/assets/4314914/afcb190f-d838-49f7-8379-76f77ac52a84)

**Local Hyrax (works)**

![hyrax4 0](https://github.com/samvera/hyrax/assets/4314914/4d57e973-7fca-4d40-b1a2-414493f0be76)

**Local Koppie (works)**

![koppie4 0](https://github.com/samvera/hyrax/assets/4314914/4b756b84-6b2e-4ecb-907a-3abfdc21fc01)

**Comment by alishaevn** [comment]
I went to the hyrax working group standup and mentioned the above issue. pg.nurax.samvera.org _IS_ equivalent to local koppie, while dev.nurax.samvera.org is equivalent to dassie.

either way, since the code is working locally for both gabriela and I, daniel wondered if it could be something up with the [nurax pg branch](https://github.com/samvera-labs/nurax/tree/pg) that deploys to pg.nurax? maybe a gem is out of date or something?

potentially unrelated, but at this moment, the pg branch is [171 commits ahead](https://github.com/samvera-labs/nurax/compare/main...pg) and [442 commits behind](https://github.com/samvera-labs/nurax/compare/pg...main) the `main` branch. daniel mentioned that there doesn't appear to be any attempt to keep them in line.

**Comment by alishaevn** [comment]
tldr: I paired with @orangewolf to get the pg app running locally and I can now duplicate the issue in the nurax app. `ActiveModel::DeprecationHandlingMessageArray` is not defined in the pg app, for some reason, while it is in hyrax.

<details>
<summary>local koppie</summary>

![image](https://github.com/samvera/hyrax/assets/29032869/96868a51-344a-4514-b029-0cbc5911142f)
</details>

<details>
<summary>local nurax</summary>

![image](https://github.com/samvera/hyrax/assets/29032869/85d93fc3-90d0-4683-b67e-dc39c022b893)
</details>

**Comment by alishaevn** [comment]
`ActiveModel::DeprecationHandlingMessageArray` is not available in [rails v6.0.6](https://api.rubyonrails.org/v6.0.6/), which is the version that the nurax `pg` branch has in its gemfile.lock. the class IS available in rails [v6.1.0](https://api.rubyonrails.org/v6.1.0/classes/ActiveModel.html), which is what the koppie app is using. (koppie is on `v6.1.7.3` to be exact)

I changed the gemfile on the nurax `pg` branch to `gem 'rails', '~> 6.0'` (what's in koppie) and `gem 'rails', '~> 6.1'`, and ran `bundle update rails --conservative`. both times I got a "Bundler attempted to update rails but its version stayed the same" message.
I even deleted the gemfile.lock file completely and ran `bundle`. it says "Bundle complete! 47 Gemfile dependencies, 326 gems now installed.", but the gemfile.lock file is still deleted in my source control, so it doesn't appear to have actually done anything.

**Comment by alishaevn** [comment]
started a [thread](https://samvera.slack.com/archives/C75LH2VEH/p1687535171894819) in the samvera nurax channel about the above issue.

**Comment by alishaevn** [comment]
@gamontoya my pr's against the pg app were deployed to pg nurax so this code is ready to be tested again! 🎉

**Comment by gamontoya** [comment]
@alishaevn I certainly appreciate your persistence--thank you!

@rjkati This ticket is now closeable. Issue resolved for pg-nurax:

- QA Testing for Hyrax
- Hyrax Version: 4.0.0
- Hyrax Branch: refs/heads/main
- Hyrax Revision: [eaf5a1f95d1d3905fbb60f5f10ca3a52c630c0a7](https://github.com/samvera/hyrax/tree/eaf5a1f95d1d3905fbb60f5f10ca3a52c630c0a7)

![Yay!](https://github.com/samvera/hyrax/assets/4314914/36b3f8a6-db1a-4e5f-9b2b-eaae4e2742ae)
